### ISSUE TITLE
Test 'about_firefox_controls' stops Iris on Mac

### ISSUE THREAD

**Issue by mwobensmith** [mozilla/iris#509]
This issue may be specific to only some MacOS versions. I am on MacOS High Sierra 10.13.4.

The test `about_firefox_controls` prevents Iris from running due to the below issue.

The keystroke combination of `Key.F2` and `KeyModifier.CTRL` causes the MacOS to minimize all windows into thumbnails, effectively killing the test and the rest of Iris' test execution. I've seen this happen, on Mac, with other key combinations that we pass to `pyautogui.hotkey` in core method `type`. 

A workaround is to press `Key.ESC` at the right moment, to force the test back into the regular windowed mode before performing additional steps. The below code works for me, which is a slight modification of what is now in `general.py,` line 260:

```
def open_about_firefox():
    if Settings.getOS() == Platform.MAC:
        # Key stroke into Firefox Menu to get to About Firefox
        type(Key.F2, modifier=KeyModifier.CTRL)
        time.sleep(0.5)
        type(Key.ESC) # added
        time.sleep(1)  # added
        type(Key.RIGHT)
        type(Key.DOWN)
        type(Key.DOWN)
        type(Key.ENTER)
```

If this problem only happens on some Macs, we may need to find out exactly why, specifically if it's a version issue. We have open issue #478 to possibly work around this, but let's find out if this affects all Macs or just some, or if the above code works universally.

If we don't have a simple solution, we will need to disable on Mac for now.

**Comment by tracywalker** [comment]
Paul, can you take a look at this?  I am on Mac 10.12, so I'll try out Matt's fix there to see if it still works for me.

**Comment by tracywalker** [comment]
The test case works fine as originally written. Matt’s change breaks the test case for me on macOS 10.12.

**Comment by tracywalker** [comment]
I'll bet I know what's going on.  in https://github.com/mozilla/iris/commit/3c0ce36216fb950d6f0c77ad1cc77889e5b462a4#diff-2318c88eb6b37be12c57f0fc640b05c0  A version of what Matt proposes above was removed from general.py helper open_about_firefox(). I can't remember exactly, but I am the one likely to have made that change to Paul's original code, so the test case would work for me on macOS 10.12.   I think, since I am the only one on 10.12, that perhaps we should standardize on 10.13. As such I upgrade to the latest.

**Comment by psilaghi** [comment]
> The test case works fine as originally checked in to master. Matt’s change breaks the test case for me on macOS 10.12.

Same for me on OS X 10.13.4.

**Comment by tracywalker** [comment]
Matt, I don't understand how using ESC works for you.  On our machines, ESC removes focus from the Main Menu (no longer highlighted) that we got by using ctrl+F2.

**Comment by psilaghi** [comment]
> The keystroke combination of Key.F2 and KeyModifier.CTRL causes the MacOS to minimize all windows into thumbnails

For me, the above combination only sets the focus to the Main Menu Bar.
Key.DOWN and KeyModifier.CTRL does minimize all windows into thumbnails for me.

**Comment by tracywalker** [comment]
It's a setting in Mac Preferences.  We all don't have the same setting(s)  Matt reassigning this to you. Paul and I have confirmed this is a settings issue, but we're unsure what differences you may have on your machine.  Let's talk when you get a chance.

**Comment by mwobensmith** [comment]
My workaround fixes it for me, but breaks it on older MacOS. Disabling test on Mac for now, will continue to look for a solution.

**Comment by tracywalker** [comment]
fixed by https://github.com/mozilla/iris/pull/550

**Comment by psilaghi** [comment]
The fix in #550 is now breaking this test intermittently.
Pressing the FN key twice causes the following popup to show up:
<img width="669" alt="screen shot 2018-06-08 at 16 46 18" src="https://user-images.githubusercontent.com/37327056/41162543-e2431446-6b3e-11e8-9ba7-258924f1548e.png">
which breaks the test.

**Comment by mwobensmith** [comment]
Sorry about that. I'm doing #595 today so this will be fixed.

**Comment by mwobensmith** [comment]
I landed #595 so this should be fixed as well.
